python/pyproject.toml

```
@@ -0,0 +1,15 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "prd"
+version = "0.0.1"  # Update as needed
+description = "Library and tools for working with PET Emission Tomography Standardization Initiative Raw Data (PETSIRD) data"
+readme = "README.md"
+authors = [{ name = "Kris Thielemans", email = "k.thielemans@ucl.ac.uk" }]
+license = { text = "Apache-2.0" }  # Update as needed
+requires-python = ">=3.8"
+dependencies = [
+    "numpy>=1.24.3",
```

`environment.yml` lists a lot of other things... are they really all for `yardl generate` and not for the end-user python package?

---

`yardl generate` doesn't need anything at all. Most of these are for building the C++ version. This is independent from the python code.

---

Ok so all the deps from `environment.yml` need to be moved here.

I'm skeptical that they really are needed though. e.g. `ninja` is only for building, not running.

Probably need something like:

```suggestion
    "numpy>=1.24"
]

[project.optional-dependencies]
cpp = ["ninja", "cmake", ...]
dev = ["twine", ...]
```

I ran `yardl generate` locally and it seems like only `numpy` is actually a dependency of the generated `python` code, though it's unclear what version is really required.

---

@johnstairs could you confirm the numpy version constraints?

---

[>= 1.22](https://microsoft.github.io/yardl/python/quickstart.html#dependencies)

---

@casperdcl do we need `cpp` for optional dependencies in `pyproject.toml`? My understanding is this would be purely for building the python version

---

continued in #37